### PR TITLE
Updating release documentation for clarity

### LINKED ISSUE #7106: Release notes are not automatically generated for the release
### Steps to reproduce

release.yaml workflow was skipped on the release PR: https://github.com/radius-project/radius/pull/7103/files due to a recent change in the workflow file.

### Observed behavior

_No response_

### Desired behavior

Release notes should be automatically generated

### Workaround

_No response_

### rad Version

0.29

### Operating system

_No response_

### Additional context

_No response_

### Would you like to support us?

- [ ] Yes, I would like to support you

[AB#11091](https://dev.azure.com/azure-octo/e61041b4-555f-47ae-95b2-4f8ab480ea57/_workitems/edit/11091)

### PR DESCRIPTION
**Pull request by willdavsmith**
# Description

* Updating release process documentation to make it more clear

## Type of change

- This pull request is a minor refactor, code cleanup, test improvement, or other maintenance task and doesn't change the functionality of Radius (issue link optional).

Closes: #7106

### CODE REVIEW

File: docs/contributing/contributing-releases/README.md
Comment (vinayada1) [inline docs/contributing/contributing-releases/README.md:50]: fix numbering

File: docs/contributing/contributing-releases/README.md
Comment (vinayada1) [inline docs/contributing/contributing-releases/README.md:44]: fix numbering

File: docs/contributing/contributing-releases/README.md
Comment (willdavsmith) [reply docs/contributing/contributing-releases/README.md:50]: I put all 1s here because markdown will do auto-numbering: https://github.com/radius-project/radius/blob/19715eea91bd1559078d1a8d496c40d977a556c9/docs/contributing/contributing-releases/README.md#creating-an-rc-release

File: docs/contributing/contributing-releases/README.md
Comment (willdavsmith) [reply docs/contributing/contributing-releases/README.md:44]: same comment as above

File: docs/contributing/contributing-releases/README.md
Comment (willdavsmith) [reply docs/contributing/contributing-releases/README.md:73]: I think this one should be version.json because it is in the bicep repo

File: .github/workflows/release.yaml
Comment (vinayada1) [inline .github/workflows/release.yaml:40]: Could we add comments in this workflow to explain the intended purpose?

**Review by vinayada1 — DISMISSED** [review]

File: docs/contributing/contributing-releases/README.md
Comment (kachawla) [inline docs/contributing/contributing-releases/README.md:123]:
```suggestion
1. In the `radius-project/radius` repo, run the [Release verification](https://github.com/radius-project/radius/actions/workflows/release-verification.yaml) workflow. Run the workflow from the release branch (format: `release/x.y`) and use the Radius release version number being released.
```

File: docs/contributing/contributing-releases/README.md
Comment (kachawla) [inline docs/contributing/contributing-releases/README.md:125]:
```suggestion
1. In the `radius-project/samples` repo, run the [Test Samples](https://github.com/radius-project/samples/actions/workflows/test.yaml) workflow. Run the workflow from the `edge` branch and using the Radius release version number being released.
```

**Review by kachawla — COMMENTED** [review]
Could you update the PR title and description to reflect a summary/description of the bug being fixed? Using 'bug from 0.30 release' is too generic and doesn't provide a clear commit message.

File: docs/contributing/contributing-releases/README.md
Comment (willdavsmith) [reply docs/contributing/contributing-releases/README.md:50]: the release branch in samples and docs branches don't exist until we do the final release, so we should run against edge until we're ready to create them.

**Review by kachawla — DISMISSED** [review]

File: docs/contributing/contributing-releases/README.md
Comment (rynowak) [inline docs/contributing/contributing-releases/README.md:41]: Should this be a fork rather than the main repo? Are we counting on the release captain to have push permissions?

File: docs/contributing/contributing-releases/README.md
Comment (rynowak) [inline docs/contributing/contributing-releases/README.md:81]: Same question, should this be a fork?

**Review by rynowak — DISMISSED** [review]
:shipit: This is a really big improvement. I had one question about whether the instructions should be talking about forks, or about the main repo. Other than that LGTM

**Review by vinayada1 — DISMISSED** [review]

**Review by kachawla — COMMENTED** [review]
@willdavsmith is this pending on addressing comments or ready for review?

**Comment by willdavsmith** [comment]
> @willdavsmith is this pending on addressing comments or ready for review?

ready for review! It was reviewed by @youngbupark during the last release and seems to be correct.

File: docs/contributing/contributing-releases/README.md
Comment (willdavsmith) [reply docs/contributing/contributing-releases/README.md:41]: Yes, for now we are counting on the release captain to have push permissions at the moment. This is because we require the GH token to create a comment on the PR. We could definitely change this but for now we should fork directly from the repo.

ref: https://github.com/radius-project/radius/issues/6927#issuecomment-1944388891

**Review by kachawla — APPROVED** [review]

**Comment by kachawla** [comment]
> > @willdavsmith is this pending on addressing comments or ready for review?
> 
> ready for review! It was reviewed by @youngbupark during the last release and seems to be correct.

great, thanks! I'll merge it once all the workflows pass.

**Pull request merged**
